File: PRNumberBiz/app/src/main/java/com/pplus/prnumberbiz/apps/setting/ui/SecessionActivity.java
package com.pplus.prnumberbiz.apps.setting.ui;

import android.content.Intent;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.fragment.app.FragmentTransaction;
import android.view.View;

import com.pple.pplus.utils.BusProvider;
import com.pplus.prnumberbiz.Const;
import com.pplus.prnumberbiz.R;
import com.pplus.prnumberbiz.apps.common.toolbar.ImplToolbar;
import com.pplus.prnumberbiz.apps.common.toolbar.OnToolbarListener;
import com.pplus.prnumberbiz.apps.common.toolbar.ToolbarOption;
import com.pplus.prnumberbiz.apps.common.ui.base.BaseActivity;
import com.pplus.prnumberbiz.apps.common.ui.base.BaseFragment;
import com.pplus.prnumberbiz.core.util.PplusCommonUtil;

/**
 * 회원탈퇴
 */
public class SecessionActivity extends BaseActivity implements ImplToolbar{

    protected ToolbarOption toolbarOption;
    private boolean leave = false;

    @Override
    public String getPID(){

        return "";
    }

    @Override
    public int getLayoutRes(){

        return R.layout.activity_secession;
    }

    public boolean isLeave(){

        return leave;
    }

    public void setLeave(boolean leave){

        this.leave = leave;
    }

    @Override
    public void initializeView(Bundle savedInstanceState){

        BusProvider.getInstance().register(this);
        BaseFragment fragment = new SecessionPreCautionFragment();

        FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
        ft.replace(R.id.secession_container, fragment, fragment.getClass().getSimpleName());
        ft.commit();
    }

    /**
     * 회원 탈퇴 Second 호출 - 인증 수단 선택
     */
    public void secessionAuth(){

    }

    public void secessionResult(){
        FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
        SecessionResultFragment fragment = SecessionResultFragment.newInstance();
        ft.replace(R.id.secession_container, fragment, SecessionResultFragment.class.getSimpleName());
        ft.commitAllowingStateLoss();
    }

    @Override
    protected void onDestroy(){

        super.onDestroy();
        BusProvider.getInstance().unregister(this);
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data){

        super.onActivityResult(requestCode, resultCode, data);
        switch (requestCode) {
            case Const.REQ_SIGN_IN:
                setResult(RESULT_OK);
                finish();
                break;
        }
    }

    @NonNull
    @Override
    public ToolbarOption getToolbarOption(){

        ToolbarOption toolbarOption = new ToolbarOption(this);
        toolbarOption.initializeDefaultToolbar(getString(R.string.word_member_leave), ToolbarOption.ToolbarMenu.LEFT);
        return toolbarOption;
    }

    @Override
    public OnToolbarListener getOnToolbarClickListener(){

        return new OnToolbarListener(){

            @Override
            public void onClick(View v, ToolbarOption.ToolbarMenu toolbarMenu, Object tag){
                switch (toolbarMenu) {
                    case LEFT:
                        if(tag.equals(1)) {
                            onBackPressed();
                        }
                        break;
                }
            }
        };
    }

    @Override
    public void onBackPressed(){
        if(leave){
            PplusCommonUtil.Companion.logOutAndRestart();
        }else{
            super.onBackPressed();
        }
    }
}

File: LuckyBol/app/src/main/java/com/pplus/luckybol/apps/setting/ui/SecessionActivity.kt
package com.pplus.luckybol.apps.setting.ui

import android.content.Intent
import android.os.Bundle
import android.view.View
import com.pplus.luckybol.Const
import com.pplus.luckybol.R
import com.pplus.luckybol.apps.common.toolbar.ImplToolbar
import com.pplus.luckybol.apps.common.toolbar.OnToolbarListener
import com.pplus.luckybol.apps.common.toolbar.ToolbarOption
import com.pplus.luckybol.apps.common.toolbar.ToolbarOption.ToolbarMenu
import com.pplus.luckybol.apps.common.ui.base.BaseActivity
import com.pplus.luckybol.databinding.ActivitySecessionBinding
import com.pplus.utils.BusProvider

/**
 * 회원탈퇴
 */
class SecessionActivity : BaseActivity(), ImplToolbar {
    override fun getPID(): String? {
        return ""
    }

    private lateinit var binding: ActivitySecessionBinding

    override fun getLayoutView(): View {
        binding = ActivitySecessionBinding.inflate(layoutInflater)
        return binding.root
    }

    override fun initializeView(savedInstanceState: Bundle?) {
        BusProvider.getInstance().register(this)
        val ft = supportFragmentManager.beginTransaction()
        ft.replace(R.id.secession_container, SecessionPreCautionFragment.newInstance(), SecessionPreCautionFragment::class.java.simpleName)
        ft.commit()
    }

    /**
     * 회원 탈퇴 Second 호출 - 인증 수단 선택
     */
    fun secessionAuth() {}
    fun secessionResult() {
        val intent = Intent(this, SecessionCompleteActivity::class.java)
        startActivity(intent)
    }

    override fun onDestroy() {
        super.onDestroy()
        BusProvider.getInstance().unregister(this)
    }

    override fun getToolbarOption(): ToolbarOption {
        val toolbarOption = ToolbarOption(this)
        toolbarOption.initializeDefaultToolbar(getString(R.string.word_member_leave), ToolbarMenu.LEFT)
        return toolbarOption
    }

    override fun getOnToolbarClickListener(): OnToolbarListener {
        return OnToolbarListener { v, toolbarMenu, tag ->
            when (toolbarMenu) {
                ToolbarMenu.LEFT -> if (tag == 1) {
                    onBackPressed()
                }
                else -> {}
            }
        }
    }
}